File: src/main/java/toby/dto/web/dnd/Feature.java
package toby.dto.web.dnd;

import com.google.gson.annotations.SerializedName;
import toby.dto.web.dnd.spell.ApiInfo;

import java.util.List;

public record Feature(String index,
                      @SerializedName("class") ApiInfo classInfo,
                      String name,
                      int level,
                      List<String> prerequisites,
                      List<String> desc,
                      String url) {
}

File: src/main/kotlin/toby/dto/web/dnd/Feature.kt
package toby.dto.web.dnd

import com.google.gson.annotations.SerializedName

data class Feature(
    val index: String?,
    @SerializedName("class")
    val classInfo: ApiInfo?,
    val name: String?,
    val level: Int?,
    val prerequisites: List<String?>,
    val desc: List<String>?,
    val url: String?
) : DnDResponse {
    override fun isValidReturnObject(): Boolean =
        !(index.isNullOrEmpty() &&
                classInfo == null &&
                name.isNullOrEmpty() &&
                level == null &&
                prerequisites.isEmpty() &&
                desc.isNullOrEmpty() &&
                url.isNullOrEmpty())
}